### PR TITLE
fix the bug of undeleted pods for tfjobs.

### PR DESCRIPTION
**Pull request by AlanFokCo**

### CODE REVIEW

File: pkg/controller.v1/tensorflow/controller.go [hunk 1]
```diff
@@ -511,7 +511,7 @@ func (tc *TFController) satisfiedExpectations(tfjob *tfv1.TFJob) bool {
 		satisfied = satisfied || tc.Expectations.SatisfiedExpectations(expectationServicesKey)
 	}
 
-	if util.CheckJobCompletedV1(tfjob.Status.Conditions) && tfjob.DeletionTimestamp == nil {
+	if util.CheckJobCompletedV1(tfjob.Status.Conditions) && tfjob.DeletionTimestamp == nil && tfjob.Annotations["arena.kubeflow.org/clean-pod-status"] == "done" {
```
Comment (cheyang) [inline pkg/controller.v1/tensorflow/controller.go]: Please define `arena.kubeflow.org/clean-pod-status` as const.

Comment (AlanFokCo) [reply pkg/controller.v1/tensorflow/controller.go]: Done

File: pkg/controller.v1/tensorflow/job.go [hunk 2]
```diff
@@ -174,6 +174,12 @@ func (tc *TFController) deletePodsAndServices(tfJob *tfv1.TFJob, pods []*v1.Pod)
 			return err
 		}
 	}
+	tfJob.Annotations["arena.kubeflow.org/clean-pod-status"] = "done"
```
Comment (cheyang) [inline pkg/controller.v1/tensorflow/job.go]: I suggest using the following style:

```golang
                tfjobToUpdate := tfjob.DeepCopy()
                ....
		tfjobToUpdate.Annotations["arena.kubeflow.org/clean-pod-status"] = "done"
		if !reflect.DeepEqual(tfjob, tfjobToUpdate) {
			_, err := tc.tfJobClientSet.KubeflowV1().TFJobs(tfjobToUpdate.Namespace).Update(tfjobToUpdate)
			if err != nil {
				return err
			}
		}
```

Comment (AlanFokCo) [reply pkg/controller.v1/tensorflow/job.go]: Done

**Review by cheyang — APPROVED** [review]

**Pull request merged**